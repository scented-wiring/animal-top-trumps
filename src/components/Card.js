import "../styles/Card.css";
import { useEffect, useState } from "react";
import { createApi } from "unsplash-js";
import logo from "../dog.png";
import key from "../accessKey";

const api = createApi({
  accessKey: key,
});

const Card = ({
  name,
  cool,
  largeness,
  handsome,
  aka,
  alignment,
  hide,
  deckSize,
  defaultText,
  message,
}) => {
  const [photo, setPhoto] = useState(null);
  const [load, setLoad] = useState(true);

  useEffect(() => {
    let isCancelled = false;
    setLoad(true);
    if (!defaultText) {
      api.search
        .getPhotos({
          query: name,
          orientation: "landscape",
          page: 1,
          perPage: 1,
        })
        .then((result) => {
          if (!isCancelled) {
            setPhoto(result.response.results[0]);
            setLoad(false);
          }
        })
        .catch(() => {
          if (!isCancelled) {
            setPhoto(false);
            setLoad(false);
          }
        });
    }
    return () => {
      isCancelled = true;
    };
  }, [name, defaultText]);

  if (defaultText) {
    return (
      <div className="Card-border">
        <div className="Card">
          <div className="default-message">{message}</div>
        </div>
      </div>
    );
  } else if (deckSize === 0) {
    return <div className="no-cards"></div>;
  } else if (hide) {
    return (
      <div className="hidden-card">
        <div id="card-back-text">
          Animal <br />
          Top Trumps
        </div>
        <img
          id="card-back-logo"
          src={logo}
          alt="Logo"
          height="96px"
          width="96px"
        />
      </div>
    );
  } else {
    return (
      <div className="Card-border">
        <div className="Card">
          <div className="name">{name}</div>
          <div className="aka">"{aka}"</div>
          <div className="alignment">({alignment})</div>
          {load ? (
            <div className="loading">Loading image...</div>
          ) : !photo ? (
            <div
              id="nophoto"
              style={{ width: "200px", height: "150px", fontSize: "20px" }}
            >
              <br />
              No photo results for "{name}" found
            </div>
          ) : (
            <img
              id="photo"
              src={photo.urls.small}
              alt="Logo"
              height="150px"
              width="200px"
            />
          )}
          <div className="stat-name1">
            Cool: <div className="stat1">{cool}</div>
          </div>
          <div className="stat-name2">
            Largeness: <div className="stat2">{largeness}</div>
          </div>
          <div className="stat-name1">
            Handsome: <div className="stat1">{handsome}</div>
          </div>
          {photo && (
            <div className="footer">
              Photo by{" "}
              <a
                className="photographer-link"
                href={`https://unsplash.com/@${photo.user.username}?utm_source=Animal Top Trumps&utm_medium=referral`}
                target="_blank"
                rel="noreferrer"
              >
                {photo.user.name}
              </a>{" "}
              on{" "}
              <a
                className="photographer-link"
                href="https://unsplash.com/?utm_source=Animal Top Trumps&utm_medium=referral"
                target="_blank"
                rel="noreferrer"
              >
                Unsplash
              </a>
            </div>
          )}
        </div>
      </div>
    );
  }
};

export default Card;
