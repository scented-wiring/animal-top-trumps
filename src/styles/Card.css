.Card-border {
  display: flex;
  flex-direction: column;
  background: #007ecc;
  border: 1px solid black;
  border-radius: 30px;
  box-shadow: 10px 10px #004742;
  height: 460px;
  width: 300px;
  padding: 10px;
}

.Card {
  position: absolute;
  height: 460px;
  width: 300px;
  background: #00bdcb;
  border-radius: 30px;
}

.name {
  font-family: "Staatliches", sans-serif;
  font-weight: bold;
  text-align: center;
  text-decoration: underline;
  font-size: 40px;
  margin: 20px;
}

.other-stat {
  font-family: "Montserrat", sans-serif;
  margin: 20px;
  font-size: 24px;
}

.delete {
  text-align: center;
}

.hidden-card {
  height: 440px;
  width: 280px;
  background: #007ecc;
  padding: 20px;
  border: 1px solid black;
  border-radius: 30px;
  display: flex;
  flex-direction: column;
  box-shadow: 10px 10px #004742;
}

.text {
  margin: 20px 0 10px 0;
  font-family: "Staatliches", sans-serif;
  font-size: 70px;
  text-align: center;
  align-items: center;
}

#card-logo {
  align-self: center;
  padding: 7px;
  border: 3px solid black;
  border-radius: 50%;
  background-color: white;
}

.no-cards {
  display: flex;
  flex-direction: column;
  height: 460px;
  width: 300px;
  padding: 10px;
}
