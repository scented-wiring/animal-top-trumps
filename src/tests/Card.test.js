import { getByRole } from "@testing-library/react";
import { render, unmountComponentAtNode } from "react-dom";
import Card from "../components/Card";

let container;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Card component", () => {
  test("renders correctly", () => {
    render(<Card />, container);
    expect(container).toMatchSnapshot();
  });

  test("renders a message if defaultText is true", () => {
    render(
      <Card message="test" alignment="do not render" defaultText={true} />,
      container
    );

    expect(container.textContent).toContain("test");
    expect(container.textContent).not.toContain("do not render");
  });

  test("renders the back of card if hide is true", () => {
    render(<Card name="test" hide={true} />, container);

    expect(container.textContent).toContain("Animal Top Trumps");
    expect(container.textContent).not.toContain("test");
  });

  test("renders nothing if decksize equals 0", () => {
    render(<Card name="test" deckSize={0} />, container);

    expect(container.textContent).not.toContain("test");
  });

  test("renders all card stats", () => {
    render(
      <Card
        name="test"
        cool="5"
        largeness="5"
        handsome="5"
        aka="test"
        alignment="test"
      />,
      container
    );

    expect(container.textContent).toContain(
      'test"test"(test)Loading image...Cool: 5Largeness: 5Handsome: 5'
    );
  });

  test("doesn't render a delete button if deleteCard prop is falsy", () => {
    render(<Card deleteCard={false} />, container);

    expect(container.textContent).not.toContain("DELETE");
  });
});
