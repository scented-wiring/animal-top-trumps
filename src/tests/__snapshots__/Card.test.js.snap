// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseCards component renders correctly 1`] = `
<div>
  <div
    class="Card"
  >
    <div
      class="name"
    />
    <div
      class="other-stat"
    >
      Cool: 
       
    </div>
    <div
      class="other-stat"
    >
      Largeness: 
       
    </div>
    <div
      class="other-stat"
    >
      Handsome: 
       
    </div>
    <div
      class="other-stat"
    >
      aka: 
      <br />
      "
      "
       
    </div>
    <div
      class="other-stat"
    >
      Alignment:
      <br />
       
    </div>
    <div
      class="delete"
    >
      <button>
        DELETE
      </button>
    </div>
  </div>
</div>
`;
